### PR TITLE
Fix URL autocomplete for useRequest

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/utils/restRequest/overrideTypes.ts [hunk 1]
```diff
@@ -52,7 +64,7 @@ type GetOption<SCHEMA, PATH extends keyof SCHEMA, OMISSION extends 'response' =
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             onFailure?: (val: TransformedError, context: any) => void;
         }, OMISSION>
-        : never
+        : unknown
```
Comment (tnagorra) [inline src/utils/restRequest/overrideTypes.ts:47]: Is this necessary?

File: src/utils/restRequest/overrideTypes.ts [hunk 2]
```diff
@@ -25,6 +25,18 @@ type ResolveBody<T> = T extends {
 
 type Callable<C, R> = R | ((value: C) => R);
 
+type GetResponse<SCHEMA, PATH extends keyof SCHEMA> = (
+    SCHEMA[PATH] extends {
+        get: {
+            parameters?: {
+                query?: UrlParams,
+                path?: AdditionalOptions['pathVariables'],
+            },
```
Comment (tnagorra) [inline src/utils/restRequest/overrideTypes.ts]: We can remove this as well.

File: src/utils/restRequest/overrideTypes.ts [hunk 3]
```diff
@@ -25,6 +25,18 @@ type ResolveBody<T> = T extends {
 
 type Callable<C, R> = R | ((value: C) => R);
 
+type GetResponse<SCHEMA, PATH extends keyof SCHEMA> = (
```
Comment (tnagorra) [inline src/utils/restRequest/overrideTypes.ts:39]: We should do these for other methods as well.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/utils/restRequest/overrideTypes.ts (hunk 1 above)
Comment (tnagorra) [reply src/utils/restRequest/overrideTypes.ts:47]: let's add a NOTE.

Comment (frozenhelium) [reply src/utils/restRequest/overrideTypes.ts:47]: Yes :(

File: src/utils/restRequest/overrideTypes.ts [hunk 4]
```diff
@@ -9,307 +8,128 @@ import {
...
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            shouldPoll?: (val: any, context: any) => number;
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            onSuccess?: (val: any, context: any) => void;
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            onFailure?: (val: TransformedError, context: any) => void;
-        }
-        : never
-);
+type CommonOption<Parameters, Responses, CONTEXT> = {
```
Comment (tnagorra) [inline src/utils/restRequest/overrideTypes.ts]:
```suggestion
type CommonOption<PARAMETERS, RESPONSES, CONTEXT> = {
```

**Review by tnagorra — APPROVED** [review]

**Pull request merged**